enterprise_access/apps/api/tasks.py

```diff
@@ -89,3 +144,177 @@ def send_notification_emails_for_requests(
+    subscription_uuid = license_assignment_results['subscription_uuid']
+
+    license_requests = LicenseRequest.objects.filter(
+        uuid__in=license_request_uuids
+    )
+
+    for license_request in license_requests:
+        user_email = learner_data[str(license_request.lms_user_id)]['email']
+
+        if not assigned_licenses.get(user_email):
+            msg = f'License was not assigned for {license_request.uuid}. {user_email} already had a license assigned.'
+            logger.info(msg)
+            license_request.state = SubsidyRequestStates.ERROR
+        else:
+            license_request.state = SubsidyRequestStates.APPROVED
+            license_request.subscription_plan_uuid = subscription_uuid
+            license_request.license_uuid = assigned_licenses[user_email]
+
+    LicenseRequest.bulk_update(license_requests, ['state', 'subscription_plan_uuid', 'license_uuid'])
+
+@shared_task(base=LoggedTaskWithRetry)
+def assign_coupon_codes_task(coupon_code_request_uuids, coupon_id):
+    """
+    Call Ecommerce API to assign coupon codes for the given coupon code requests.
+
+    Args:
+        coupon_code_request_uuids (list of UUID): list of coupon code request UUIDs to assign coupon codes for
+        coupon_id (int): the id of the coupon to assign codes from
+
+    Returns:
+        A dict representing coupon code assignment results in the form of:
+            {
+                'coupon_code_request_uuids' (list of UUID): coupon code request UUIDs that were processed,
+                'learner_data' (dict): dict containing user data keyed by lms_user_ids,
+                'assigned_codes' (dict): dict containing codes assigned keyed by lms_user_ids,
+                'coupon_id' (int): the id of the coupon that codes were assigned from
+            }
+    """
+
+    coupon_code_requests = CouponCodeRequest.objects.filter(
+        uuid__in=coupon_code_request_uuids,
+        state=SubsidyRequestStates.PENDING
+    )
+
+    if not coupon_code_requests:
+        logger.info(f'No pending/errored coupon code requests with uuids: {coupon_code_requests} found.')
+        return None
+
+    lms_user_ids = [request.lms_user_id for request in coupon_code_requests]
+    learner_data = _get_enterprise_learner_data(lms_user_ids)
+    user_emails = [user['email'] for user in learner_data.values()]
+
+    ecommerce_api_client = EcommerceApiClient()
+    response = ecommerce_api_client.assign_coupon_codes(user_emails, coupon_id)
+    assigned_codes = { assignment['user_email']: assignment['code'] for assignment in response['offer_assignments'] }
+
+    return {
+        'coupon_code_request_uuids': coupon_code_request_uuids,
+        'learner_data': learner_data,
+        'assigned_codes': assigned_codes,
+        'coupon_id': coupon_id
+    }
+
+
+@shared_task(base=LoggedTaskWithRetry)
+def update_coupon_code_requests_after_assignments_task(coupon_code_assignment_results):
```

Comment:
for this and the other "_after_assignments_task" -- the functionality is fine, but i personally don't love this function's name "knowing" where it's going to get called, because there's technically nothing special about being called after the other task

Reply:
leaving for now since removing _after_assignments_task makes it too general.